common/src/main/java/com/wynntils/features/user/overlays/NpcDialogueOverlayFeature.java

```diff
@@ -75,13 +80,32 @@ private void cancelAutoProgress() {
 
     @SubscribeEvent(priority = EventPriority.HIGHEST)
     public void onNpcDialogue(NpcDialogEvent e) {
-        String msg = e.getChatMessage() == null ? null : ComponentUtils.getCoded(e.getChatMessage());
-        if (msg != null && NEW_QUEST_STARTED.matcher(msg).find()) {
+        List<String> msg =
+                e.getChatMessage().stream().map(ComponentUtils::getCoded).toList();
+        if (e.getType() == NpcDialogueType.CONFIRMATIONLESS && !currentDialogue.isEmpty()) {
+            // Now we have two concurrent dialogues, so combine the two into one instead of
+            // just replacing the old.
+            ArrayList<String> combinedDialogue = new ArrayList<>(currentDialogue);
+            combinedDialogue.add("");
+            combinedDialogue.addAll(msg);
+            currentDialogue = combinedDialogue;
+            if (dialogueType == NpcDialogueType.CONFIRMATIONLESS) {
+                // If it was confirmationless, delay it's removal
+                removeTime = System.currentTimeMillis() + calculateMessageReadTime(msg);
+            }
+        } else {
+            currentDialogue = msg;
+            dialogueType = e.getType();
+            if (dialogueType == NpcDialogueType.CONFIRMATIONLESS) {
+                removeTime = System.currentTimeMillis() + calculateMessageReadTime(msg);
+            }
```

Comment:
This part seems to be common? Extract it?

Reply:
Yeah, about that. Code quality. It shucks. Big time. Not up to my usual standard. :-( I was hoping to have a Chat Manager update that would leave it shining and sparkling bright and easy to understand. Instead there were just more special handling, convoluted code and strange edge cases. :-(

I have pretty high confidence in the code, since I've tested it thoroughly, and also played a real bit of the "All Roads to Peace" quest, where there are a lot of odd dialogs with the Orcs.

But it looks like shite. 

I would like to improve this, but I don't know how, and I'm worried that I might break things if I poke around too much.

But I agree that the duplication sticks in your eyes. I'll see if I can do something about that.

Reply:
Oh, I did not see where you commented! :) I missed this part, fixed now. I thought you were talking about the massive chunk of duplication in the ChatHandler after `  } else if (NPC_SELECT_PATTERN.matcher(firstLineCoded).find()) {`... I am not proud of that. I'll look if I can make something about it now that I'm not half falling asleep.